### ISSUE TITLE
Submitting form works on iOS and web, but not on Android

### ISSUE THREAD

**Issue by peterlauri** [flutter/flutter#48513]
Very simple login form, works on iOS and web, but not on Android. Is there something fundamental, or is there some bug related to this. Basically, clicking on the "submit" button does not make make the login call.

```dart
import 'package:flutter/material.dart';
import 'package:flutter_poc/app_state.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' as convert;

class AuthPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Center(child: MyCustomForm()),
      ),
    );
  }
}

class MyCustomForm extends StatefulWidget {
  @override
  MyCustomFormState createState() {
    return MyCustomFormState();
  }
}

class MyCustomFormState extends State<MyCustomForm> {
  final _formKey = GlobalKey<FormState>();
  String _username;
  String _password;

  Future<String> getAuthToken(username, password) async {
    final postData = {'username': username, 'password': password};
    final loginUrl = 'https://xxx/login';
    final response = await http.post(loginUrl, body: postData);
    if (response.statusCode == 200) {
      final data = convert.jsonDecode(response.body);
      return data['token'];
    }
    return null;
  }

  @override
  Widget build(BuildContext context) {
    // Build a Form widget using the _formKey created above.
    return Form(
      key: _formKey,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: <Widget>[
          TextFormField(
            decoration: InputDecoration(hintText: 'Email'),
            validator: (value) {
              if (value.isEmpty) {
                return 'Enter email';
              }
              return null;
            },
            onSaved: (val) => {_username = val},
          ),
          TextFormField(
            decoration: InputDecoration(hintText: 'Password'),
            validator: (value) {
              if (value.isEmpty) {
                return 'Enter password';
              }
              return null;
            },
            obscureText: true,
            onSaved: (val) => {_password = val},
          ),
          Padding(
            padding: const EdgeInsets.symmetric(vertical: 16.0),
            child: RaisedButton(
              onPressed: () async {
                final form = _formKey.currentState;
                if (form.validate()) {
                  form.save();
                  final token = await getAuthToken(_username, _password);
                  if (token == null) {
                    Scaffold.of(context).showSnackBar(
                        SnackBar(content: Text('failed to login')));
                  } else {
                    AppState().setToken(token);
                  }
                }
              },
              child: Text('Submit'),
            ),
          ),
        ],
      ),
    );
  }
}

```

```console
➜  spine git:(develop) flutter doctor -v
[✓] Flutter (Channel beta, v1.12.13+hotfix.6, on Mac OS X 10.15.2 19C57, locale en-GB)
    • Flutter version 1.12.13+hotfix.6 at /Users/plauri/.applications/flutter
    • Framework revision 18cd7a3601 (4 weeks ago), 2019-12-11 06:35:39 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/plauri/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (3 available)
    • iPhone 11  • 6C358D02-1F75-4E8D-AE23-8E2257246021 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
    • Chrome     • chrome                               • web-javascript • Google Chrome 79.0.3945.88
    • Web Server • web-server                           • web-javascript • Flutter Tools

! Doctor found issues in 1 category.
```

**Comment by iapicca** [comment]
Hi @peterlauri 
can you please provide your `flutter run --verbose`?
Thank you

**Comment by peterlauri** [comment]
@iapicca this issue can be closed. I didn't notice the problem directly, as I never got any direct response, so always rebooted. But now when I ran in verbose mode I took a cup of coffee, and noticed that there were a DNS resolution error. Basically the Android emulator that I used did not pick up the DNS from the host machine, so it could not resolve the api address. Solved my problem by setting dns manually, like here: https://stackoverflow.com/questions/44535500/internet-stopped-working-on-android-emulator-mac-os
